health/service/src/main/java/com/cerner/beadledom/health/internal/HealthChecker.java

```diff
@@ -56,8 +57,13 @@
    * Other metadata fields on the HealthDto will also be populated.
    */
   public HealthDto doPrimaryHealthCheck() {
-    // TODO - allow some dependencies to be marked as 'secondary', and skip those here
-    return checkHealth(healthDependencies.values());
+    List<HealthDependency> dependencies = new ArrayList<>();
```

Comment:
You can use the stream api,

```
List<HealthDependency> primaryHealthDependencies = healthDependencies.values().stream()
        .filter(HealthDependency::getPrimary)
        .collect(Collectors.toList());
```

Another option would be to create to create two maps in the constructor, one with the primary and the other with the ancillary dependencies.

Reply:
Changed to use stream api: https://github.com/nab0310/beadledom/commit/4928608b1f95af3516d4c682c8c88b17c0832d15